Tests/FunctionalTests/AEPEdgeResponseHandlerFunctionalTests.swift

```diff
@@ -0,0 +1,192 @@
+                                                             response: HTTPURLResponse(url: edgeUrl,
+                                                                                       statusCode: 200,
+                                                                                       httpVersion: nil,
+                                                                                       headerFields: nil),
+                                                             error: nil)
+        let httpConnection2: HttpConnection = HttpConnection(data: responseBodyWithTwoErrors.data(using: .utf8),
+                                                             response: HTTPURLResponse(url: edgeUrl,
+                                                                                       statusCode: 200,
+                                                                                       httpVersion: nil,
+                                                                                       headerFields: nil),
+                                                             error: nil)
+        let responseHandler1 = MockResponseHandler(expectedErrors: 2)
+        let responseHandler2 = MockResponseHandler(expectedResponses: 1)
+
+        // set expectations & send two events
+        setExpectationNetworkRequest(url: FunctionalTestConst.EX_EDGE_INTERACT_URL_STR, httpMethod: HttpMethod.post, expectedCount: 1)
+        setNetworkResponseFor(url: FunctionalTestConst.EX_EDGE_INTERACT_URL_STR, httpMethod: HttpMethod.post, responseHttpConnection: httpConnection1)
+        Edge.sendEvent(experienceEvent: ExperienceEvent(xdm: ["eventType": "personalizationEvent", "test": "xdm"],
+                                                        data: nil),
+                       responseHandler: responseHandler1)
+        assertNetworkRequestsCount()
+        responseHandler1.await()
+
+        resetTestExpectations()
+        setExpectationNetworkRequest(url: FunctionalTestConst.EX_EDGE_INTERACT_URL_STR, httpMethod: HttpMethod.post, expectedCount: 1)
+        setNetworkResponseFor(url: FunctionalTestConst.EX_EDGE_INTERACT_URL_STR, httpMethod: HttpMethod.post, responseHttpConnection: httpConnection2)
+        Edge.sendEvent(experienceEvent: ExperienceEvent(xdm: ["eventType": "personalizationEvent", "test": "xdm"],
+                                                        data: nil),
+                       responseHandler: responseHandler2)
+
+        // verify
+        assertNetworkRequestsCount()
+        responseHandler2.await()
+
+        XCTAssertEqual(1, responseHandler1.onResponseHandles.count)
+        XCTAssertEqual(0, responseHandler1.onErrorHandles.count)
+        XCTAssertTrue(responseHandler1.onCompleteCalled)
+        XCTAssertEqual(0, responseHandler2.onResponseHandles.count)
+        XCTAssertEqual(2, responseHandler2.onErrorHandles.count)
+        XCTAssertTrue(responseHandler2.onCompleteCalled)
```

Comment:
Are the response handlers reversed here?  You initialize responseHandler1 with 2 expected errors but assert that there is 1 response handler and 0 error handlers. Similarly, responseHandler2 is initialized with 1 expected response handler but you assert it has 2 error handlers and 0 response handlers.